### PR TITLE
SCAL-75674-param-based-changes-in-docs

### PR DESCRIPTION
**Pull request by pankajp-ts**
Ticket - https://thoughtspot.atlassian.net/browse/SCAL-75674

- example url to test `https://ts-blink.github.io/embed-sdk/en/?link=https://thoughtspot.com&value=dynamic-override`

### CODE REVIEW

File: docs/src/utils/app-utils.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const queryStringParser = (queryParamStr) => {
+    const queryParamObj = {};
+    if (!queryParamStr) return queryParamObj;
+
+    const entries = new URLSearchParams(queryParamStr).entries();
```
Comment (ayonghosh) [inline docs/src/utils/app-utils.js:5]: URLSearchParams don't work on IE. Maybe we should include a polyfill as well?

File: docs/src/utils/app-utils.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export const queryStringParser = (queryParamStr) => {
```
Comment (ayonghosh) [inline docs/src/utils/app-utils.js:1]: Let's add a jsDoc comment on what this function does, input and output

File: docs/src/utils/doc-utils.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+export const htmlParser = (html, params) => {
```
Comment (ayonghosh) [inline docs/src/utils/doc-utils.js:1]: We should rename this function, we're not parsing HTML here but substituting params in the content

File: docs/src/utils/doc-utils.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    let parsedHtml = html;
+    const paramKeys = Object.keys(params);
+    if (!html && paramKeys.length === 0) return parsedHtml;
+    const customPassThroughStart = '{{';
+    const customPassThroughEnd = '}}';
+
+    paramKeys.map((key) => {
+        console.log('key = ', key, params[key]);
```
Comment (ayonghosh) [inline docs/src/utils/doc-utils.js:9]: Remove console statement

File: scripts/gatsby-commands.js [hunk 5]
```diff
@@ -15,6 +15,30 @@ args.splice(0, 2);
 // used to handle command options supplied as arguments
 const cmdOptionHandler = (command, options) => {
     switch (command) {
+        case 'develop': {
+            const cmdOptions = options[1] && options[1].split('=');
```
Comment (ayonghosh) [inline scripts/gatsby-commands.js:19]: Consider using this: https://www.npmjs.com/package/command-line-args

File: scripts/gatsby-commands.js [hunk 6]
```diff
@@ -2,114 +2,164 @@
...
-switch (cmdName) {
-    case 'develop':
-        cmdToExecute = `gatsby develop`;
+switch (mainOptions.command) {
+    case 'develop': {
+        const developDefinitions = [
+            {
+                name: 'port',
+                alias: 'p',
+                type: Number,
```
Comment (ayonghosh) [inline scripts/gatsby-commands.js:89]: Use the lower case number

File: scripts/gatsby-commands.js [hunk 7]
```diff
@@ -2,114 +2,164 @@
...
         break;
-    case 'build':
-        const prefixOption = cmdOptionHandler(cmdName, args);
-        if (!prefixOption) return;
+    }
+    case 'build': {
+        const buildDefinitions = [
+            {
+                name: 'noprefix',
+                type: Boolean,
```
Comment (ayonghosh) [inline scripts/gatsby-commands.js:108]: use the lowercase boolean

File: docs/src/pages/index.js [hunk 8]
```diff
@@ -1,10 +1,27 @@
-import * as React from 'react';
```
Comment (ayonghosh) [inline docs/src/pages/index.js:1]: Is it possible to use TypeScript here? I'm guessing Gatsby would support TS

**Review by ayonghosh — APPROVED** [review]
Let's switch to TypeScript in the next PR

**Comment by pankajp-ts** [comment]
> Let's switch to TypeScript in the next PR

Sure.

**Pull request merged**